src/main/java/com/puppycrawl/tools/checkstyle/utils/UnmodifiableCollectionUtil.java

```diff
@@ -63,6 +67,20 @@ public static <T> List<T> unmodifiableList(List<T> collection) {
         return Collections.unmodifiableList(collection);
     }
 
+    /**
+     * Returns an unmodifiable view of a List containing elements of a specific type.
+     *
+     * @param items The List of items to make unmodifiable.
+     * @param elementType The Class object representing the type of elements in the list.
+     * @param <T> The type of elements in the resulting unmodifiable List.
+     * @return An unmodifiable List containing elements of the specified type.
+     */
+    public static <T> List<T> unmodifiableList(Collection<Item> items, Class<T> elementType) {
```

Comment:
Why do you use exact type here (which is `Item`). This method is supposed to be generic. Can we update it as follows?
```
public static <S, T> List<T> unmodifiableList(Collection<S> items, Class<T> elementType) { ... }
```
here, `S` means source type and `T` means target type.

Reply:
@Kevin222004 , please fix as suggested.

Reply:
done

Reply:
Do we need this block after the above change? Could you remove it if not needed?
```
    <file name="UnmodifiableCollectionUtil">
       <allow pkg="net.sf.saxon"/>
     </file>
```

Reply:
@rdiachenko removed.